test/integration/listener_lds_integration_test.cc

```diff
@@ -821,10 +821,18 @@ TEST_P(ListenerIntegrationTest, RemoveListenerAfterInPlaceUpdate) {
 
   // All the listen socket are closed. include the sockets in the active listener and
   // the sockets in the filter chain draining listener. The new connection should be reset.
-  auto codec =
+  auto codec1 =
       makeRawHttpConnection(makeClientConnection(lookupPort(listener_name_)), absl::nullopt);
-  EXPECT_FALSE(codec->connected());
-  EXPECT_THAT(codec->connection()->transportFailureReason(), StartsWith("delayed connect error"));
+  // The socket are closed asynchronously, so waiting the connection closed here.
+  while (codec1->connected()) {
+    dispatcher_->run(Event::Dispatcher::RunType::NonBlock);
```

Comment:
can this be done with a waitFor variant? It looks like this pattern will spin. Maybe waitForDisconnect in integration_tcp_client.cc ?

there's one below as well.

Reply:
yea, waitForDisconnect is right thing, thanks!